MdeModulePkg/Core/Pei/Ppi/Ppi.c

```diff
@@ -324,16 +324,9 @@ ConvertPpiPointersFv (
 
     Guid = PrivateData->PpiData.PpiList.PpiPtrs[Index].Ppi->Guid;
     for (GuidIndex = 0; GuidIndex < ARRAY_SIZE (GuidCheckList); ++GuidIndex) {
-      //
-      // Don't use CompareGuid function here for performance reasons.
-      // Instead we compare the GUID as INT32 at a time and branch
-      // on the first failed comparison.
-      //
-      if ((((INT32 *)Guid)[0] == ((INT32 *)GuidCheckList[GuidIndex])[0]) &&
-          (((INT32 *)Guid)[1] == ((INT32 *)GuidCheckList[GuidIndex])[1]) &&
-          (((INT32 *)Guid)[2] == ((INT32 *)GuidCheckList[GuidIndex])[2]) &&
-          (((INT32 *)Guid)[3] == ((INT32 *)GuidCheckList[GuidIndex])[3]))
-      {
+      // MU_CHANGE [BEGIN] - CodeQL change
+      if (CompareMem (Guid, GuidCheckList[GuidIndex], 16) == 0) {

```

Comment:
Like the other PR, I'd use `CompareGuid()` here if there's not a known reason to avoid it.

Reply:
No reason so I changed it.